packages/contracts/src/dollar/StakingShare.sol

```diff
@@ -214,15 +217,29 @@ contract StakingShare is
         uint256[] memory ids,
         uint256[] memory amounts,
         bytes memory data
-    ) internal virtual override(ERC1155SetUri, ERC1155PausableSetUri) {
+    ) internal virtual override(ERC1155, ERC1155Pausable) {
         super._beforeTokenTransfer(operator, from, to, ids, amounts, data);
     }
 
+    function uri(
+        uint256 tokenId
+    ) public view virtual override(ERC1155, ERC1155URIStorage) returns (string memory) {
+        return _uri;
```

Comment:
This method returns `uri` from the `StakingShare` contract. If we call `setUri(1, "https://site.com/tokens/1.json")` then `uri(1)` will return `_uri` from the `StakingShare` contract, but it is expected to return `https://site.com/tokens/1.json`.

You can refactor the code with the following methods:
```
    function uri(
        uint256 tokenId
    ) public view virtual override(ERC1155, ERC1155URIStorage) returns (string memory) {
        return super.uri(tokenId);
    }

    function setBaseUri(string memory newUri) external onlyStakingManager {
        _setBaseURI(newUri);
    }

    function setUri(uint256 tokenId, string memory tokenUri) external onlyStakingManager {
        _setURI(tokenId, tokenUri);
    }

    function setUri(string memory tokenUri) external onlyStakingManager {
        _setURI(tokenUri);
    }
```

This way we have the ability to update URI for the `ERC1155` contract as well as URIs for single tokens which might be useful in the future.

So please update the code along with tests.

Reply:
This is cool. Code updated